README updates (#46)
signalfx/splunk-otel-collector: README.md

- `SPLUNK_MEMORY_LIMIT_MIB` (no default): Maximum total memory to be allocated by the process heap.
- `SPLUNK_MEMORY_SPIKE_MIB` (no default): Maximum spike between the measurements of memory usage.

> Deploy the collector as outlined in the below.

Reviewer: ```suggestion
The following sections describe how to deploy the Collector in supported environments.
```

### Docker

> Deploy from a Docker container (replace `0.1.0` with the latest stable version number):

Reviewer: ```suggestion
Deploy from a Docker container. Replace `0.1.0` with the latest stable version number:
```

is there a reason we specify 0.1.0 here? can we replace `0.1.0` with `<version>` and tell users to replace it with the latest stable version instead?

Author: Could do it though one is a valid config and other is not. Currently like this as it follows product docs: https://docs.signalfx.com/en/latest/apm/apm-getting-started/apm-opentelemetry-collector.html.

### Linux Installer Script

For non-containerized Linux environments, a convenience script is available for

> installing the collector package and [TD Agent

Reviewer: ```suggestion
installing the Collector package and [TD Agent
```

(Fluentd)](https://www.fluentd.org/).

> Run the following command on your host (replace `SPLUNK_REALM`,
> `SPLUNK_BALLAST_SIZE`, and `SPLUNK_ACCESS_TOKEN` appropriately for your
> environment):

Reviewer: ```suggestion
Run the following command on your host. Replace `SPLUNK_REALM`,
`SPLUNK_BALLAST_SIZE`, and `SPLUNK_ACCESS_TOKEN` for your
environment:
```

```sh
curl -sSL https://dl.signalfx.com/splunk-otel-collector.sh > /tmp/splunk-otel-collector.sh;
sudo sh /tmp/splunk-otel-collector.sh --realm SPLUNK_REALM --ballast SPLUNK_BALLAST_SIZE \
    -- SPLUNK_ACCESS_TOKEN
```

You can view the [source](internal/buildscripts/packaging/installer/install.sh)
for more details and other options.

Currently, only the following Linux distributions and versions are supported:

- Amazon Linux: 2
- CentOS / Red Hat / Oracle: 7, 8
- Debian: 8, 9, 10
- Ubuntu: 16.04, 18.04, 20.04

## Sizing

The OpenTelemetry Collector can be scaled up or out as needed. Sizing is based
on the amount of data per data source and requires 1 CPU core per:

- Traces: 10,000 spans per second
- Metrics: 20,000 data points per second

> > If a Collector handles both trace and metric data then both must be accounted
> > for when sizing. For example, 5K spans per second plus 10K data points per
> > second would require 1 CPU core.

Reviewer: ```suggestion
If a Collector handles both trace and metric data then both must be accounted
for when sizing. For example, 5K spans per second plus 10K data points per
second would require 1 CPU core.
```

The recommendation is to use a ratio of 1:2 for CPU:memory and to allocate at
least a CPU core per Collector. Multiple Collectors can deployed behind a
simple round-robin load balancer. Each Collector runs independently, so scale
increases linearly with the number of Collectors you deploy.

> > The Collector does not persist data to disk so no disk space is required.

Reviewer: ```suggestion
The Collector does not persist data to disk so no disk space is required.
```
let's move this up to the first paragraph of this section.

$ docker run --rm -e SPLUNK_ACCESS_TOKEN=12345 -e SPLUNK_BALLAST_SIZE_MIB=683 \
    -e SPLUNK_REALM=us0 -p 13133:13133 -p 14250:14250 -p 14268:14268 -p 55678-55680:55678-55680 \
    -p 6060:6060 -p 7276:7276 -p 8888:8888 -p 9411:9411 -p 9943:9943 \
    --name otelcol quay.io/signalfx/splunk-otel-collector:0.1.0 \
        --log-level=DEBUG
```

> > Use `--help` to see all available CLI arguments.

Reviewer: ```suggestion
Use `--help` to see all available CLI arguments.
```

### Custom Configuration

In addition to using the default configuration, a custom configuration can also

> be provided. This can be done via the `SPLUNK_CONFIG` environment variable as
> well as the `--config` command line argument.

Reviewer: ```suggestion
be provided. Use the `SPLUNK_CONFIG` environment variable or
the `--config` command line argument to provide a custom configuration.
```

> > Command line arguments take precedence over environment variables. This
> > applies to `--config` and `--mem-ballast-size-mib`.

Reviewer: ```suggestion
Command line arguments take precedence over environment variables. This
applies to `--config` and `--mem-ballast-size-mib`.
```

Author: I prefer for call outs to be clear to see in the readme and thus the blockquote
